src/rpc_server.rs

```diff
@@ -3129,6 +3234,43 @@ impl RPC for NeptuneRPCServer {
         Ok(Self::cpu_temp_inner())
     }
 
+    // documented in trait. do not add doc-comment.
+    async fn pow_challenge(
+        self,
+        _context: tarpc::context::Context,
+        token: rpc_auth::Token,
+    ) -> RpcResult<Option<GuessChallengeInfo>> {
+        log_slow_scope!(fn_name!());
+        token.auth(&self.valid_tokens)?;
+
+        let Some(proposal) = self
+            .state
+            .lock_guard()
+            .await
```

Comment:
The lock guard is acquired multiple times. Can this be factored out into a variable?